config/jobs/kubernetes/test-infra/test-infra-trusted.yaml

```diff
@@ -793,13 +793,18 @@ periodics:
     - name: gcloud
       image: gcr.io/k8s-staging-test-infra/gcloud-in-go:v20230111-cd1b3caf9c
       command:
-      - bash
-        - -c
-        -
-            temp_file="$(mktemp)"
-            && gcloud iam service-accounts keys create "${temp_file}" --iam-account=pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com --key-file-type=json
-            && gcloud secrets versions add default-k8s-build-cluster-service-account-key --data-file=${temp_file} --project=k8s-prow-builds
-            && for key_id in $(gcloud iam service-accounts keys list --iam-account=pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com --managed-by=user --created-before=$(date +%Y-%m-%d -d "90 days ago") --format="value(KEY_ID)"); do gcloud iam service-accounts keys delete "$key_id" -q --iam-account=pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com;done
+      - /bin/bash
+      args:
+      - -c
+      - |
+        set -euo pipefail
+
+        temp_file="$(mktemp)"
```

Comment:
I think to do a `-c` like this and not use `&&` you need to put `;` at the end of each command so it's actually interpreted as multiple commands. I don't think `-c` respects newlines

Reply:
/hold

Reply:
not important: but we might want to `trap 'rm -f "${temp_file:?}"' EXIT` to avoid the temp file being persisted on completed but not yet GCed pods /shrug

Reply:
But see the known-good-example https://github.com/kubernetes/test-infra/blob/3c77b9b170e6605d17e9ca021a4c2164f26eb10a/config/jobs/kubernetes/sig-arch/kubernetes-depstat-periodical.yaml#L16-L31

(Link was mentioned in my commit message)

Reply:
/hold cancel